### PR TITLE
Registration login fancy

### LINKED ISSUE #669: Registration form shouldn't request a username
There's some discussion about this in #614. The user registration form currently requires users to decide on a username when creating an account; it shouldn't.

### LINKED ISSUE #671: User: Registration/login form
- remove gross default helper text
- add placeholder values
- updated form titles

### PR DESCRIPTION
**Pull request by angaither**
Add placeholder text to login/registration.
Remove default helper text, bleh. :gun: :floppy_disk:
Hide username field. 
Automagically generate usernames (it's a UID) :bomb: 

Fixes #669
Fixes #671

### CODE REVIEW

**Comment by DFurnes** [comment]
Looks beauteous to me. :dancer:

File: lib/modules/dosomething/dosomething_user/dosomething_user.module [hunk 1]
```diff
@@ -105,12 +112,79 @@ function dosomething_user_form_alter(&$form, $form_state, $form_id) {
...
+ *  A drupal form_state array.
+ *
+ * @return - array - $form
+ *  Modified drupal form.
+ */
+function dosomething_user_register_after_build($form, &$form_state) {
+  unset($form['field_birthdate'][LANGUAGE_NONE][0]['value']['date']['#title']);
+  unset($form['field_birthdate'][LANGUAGE_NONE][0]['value']['date']['#description']);
+  $form['field_birthdate'][LANGUAGE_NONE][0]['value']['date']['#attributes']['placeholder'] = t('MM/DD/YYYY');
+  $form['field_first_name'][LANGUAGE_NONE][0]['value']['#attributes']['placeholder'] = t('[What do we call you?]');
```
Comment (DFurnes) [inline lib/modules/dosomething/dosomething_user/dosomething_user.module:169]: What's the dealio with the brackets around the placeholder text?

Comment (angaither) [inline lib/modules/dosomething/dosomething_user/dosomething_user.module:169]: ehh, I copy and pasted from what was in new world prototype stuff.

Comment (DFurnes) [inline lib/modules/dosomething/dosomething_user/dosomething_user.module:169]: Alright, I'm going to switch those over to parenthesis but I can do that in a separate PR.

**Comment by DFurnes** [comment]
@angaither Is this ready to merge in? :dancers:

**Comment by aaronschachter** [comment]
:+1:

**Pull request merged**